drizzlepac/alignimages.py

```diff
@@ -202,361 +197,480 @@ def perform_align(input_list, **kwargs):
-        for item in imglist:
-            item.meta.update(item.best_meta)
-        filteredTable['status'][:] = 0
-        fitStatusDict = best_fitStatusDict.copy()
-
-        # Protect the writing of the table within the best_fit_rms
-        info_keys = OrderedDict(imglist[0].meta['fit_info']).keys()
-        # Update filtered table with number of matched sources and other information
-        for item in imglist:
-            imgname = item.meta['name']
-            index = np.where(filteredTable['imageName'] == imgname)[0][0]
-
-            if not item.meta['fit_info']['status'].startswith("FAILED"):
-                for tweakwcs_info_key in info_keys:
-                    if not tweakwcs_info_key.startswith("matched"):
-                        if tweakwcs_info_key.lower() == 'rms':
-                            filteredTable[index]['rms_x'] = item.meta['fit_info'][tweakwcs_info_key][0]
-                            filteredTable[index]['rms_y'] = item.meta['fit_info'][tweakwcs_info_key][1]
-
-                filteredTable[index]['catalog'] = item.meta['fit_info']['catalog']
-                filteredTable[index]['catalogSources'] = len(reference_catalog)
-                filteredTable[index]['matchSources'] = item.meta['fit_info']['nmatches']
-                filteredTable[index]['rms_ra'] = item.meta['fit_info']['RMS_RA'].value
-                filteredTable[index]['rms_dec'] = item.meta['fit_info']['RMS_DEC'].value
-                filteredTable[index]['fit_rms'] = item.meta['fit_info']['FIT_RMS']
-                filteredTable[index]['total_rms'] = item.meta['fit_info']['TOTAL_RMS']
-                filteredTable[index]['offset_x'], filteredTable[index]['offset_y'] = item.meta['fit_info']['shift']
-                filteredTable[index]['scale'] = item.meta['fit_info']['scale'][0]
-                filteredTable[index]['rotation'] = item.meta['fit_info']['rot']
-
-                # populate filteredTable fields "status", "compromised" and
-                # "processMsg" with fitStatusDict fields "valid", "compromised"
-                # and "reason".
-                explicitDictKey ="{},{}".format(item.meta['name'], item.meta['chip'])
-                if fitStatusDict[explicitDictKey]['valid'] == True:
-                    filteredTable[index]['status'] = 0
+            extracted_sources = generate_source_catalogs(process_list,
+                                                         centering_mode='starfind',
+                                                         nlargest=num_sources,
+                                                         output=output)
+
+        for imgname in extracted_sources.keys():
+            table = extracted_sources[imgname]["catalog_table"]
+
+            # Get the location of the current image in the filtered table
+            index = np.where(filtered_table['imageName'] == imgname)[0][0]
+
+            # First ensure sources were found
+
+            if table is None or table[1] == None: 
```

Comment:
```suggestion
            # This logic is actually correct, since 'table={1:None}' if no sources are found.
            if table[1] == None:
```

Reply:
I would say leave the code alone "as is" if it works for you. I do not understand the way the data are organized and so I may have missed something. If `table={1:None}` for the case of no sources, then I can't understand why my initial suggestion did not work:
```python
>>> a = {1:None}                                                                                                           
>>> a[1] is None                                                                                                           
True
>>> a[1] == None                                                                                                           
True
```
Most likely, however, (I feel) you have empty `astropy.table.Table` and if that is the case, you should change the code to:
```python
if table is None or not table[1]:
```
This last suggestion will work clean when `table[1]` is either `None` or an empty table. Comparing a table to `None` to check its length is _very_ unusual. (For example if `a=[]` I would do `if not a:` and not `if a == None:`)